great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -191,6 +191,62 @@
 T = TypeVar("T", dict, list, str)
 
 
+class DatasourceDict(UserDict):
+    def __init__(self, context: AbstractDataContext):
+        self._context = context
+        self._datasource_store = context._datasource_store
+        self._config_provider = context.config_provider
+        self._fluent_config = context.fluent_config
+
+    @property
+    def data(self):
+        keys = self._datasource_store.list_keys()
+        names = [key.resource_name for key in keys]
+        return {name: self.__getitem__(name) for name in names}
```

Comment:
`UserDict` relies on an underlying `data` attr - while I can (and have) implemented specific dunder methods, writing up an impl for `data` ensures that the parent class methods are functional